guides/ADDING_A_NEW_PRODUCT.md

```diff
@@ -0,0 +1,160 @@
+- workshop defined in `allWorkshopStages`
+
+### Deciding when to create a new stage
+New stages are not desirable, as the operator will not progress to the following phase, until everything in the current
+phase has reported that it has completed, so adding new stages can slow down installation time.
+
+The idea of a stage is to allow the operator to complete some action, that will be required in the following stage, if 
+product A is required by product B, product A should probably go in an earlier stage than product B. For example, RHSSO
+requires cloud resource, so the cloud resources operator had to go into it's own stage, prior to the stage that RHSSO is 
+in.
+
+In general, if the new product is another tool for developers to use, it probably belongs in the products stage. Where 
+you add it inside a stage is immaterial; as these are not processed in any particular order, and nothing in a stage 
+should depend on something else in the same stage.
+
+## New Product Reconciler
+The reconciler must implement the `Products.Interface` interface, in order to work with  the installation controller. 
+The methods are defined in more detail below:
+
+### Reconcile
+This is the primary entry point of your reconciler, on each reconcile loop of the integreatly-operator,
+all the logic in here should be written with the assumption that resources may or may not exist, and can
+be created or updated based on their current state. 
+
+As a side-note, it's worth keeping in mind that there are quite a lot of helpful functions in the resource package for 
+handling the more common tasks, such as setting up the CSV, catalog source, subscription and namespaces. Many examples 
+of this are available in the other reconcilers (such as Code-Ready, RHSSO, etc). 
+
+There is also a helper function to add a finalizer to the RHMI CR, this also requires a function allowing the organisation 
+of the tear-down of the product at uninstall time, again examples of this can are available in the existing reconcilers.
+
+#### Parameters
+
+There are several parameters passed into this function:
+##### ctx
+ This must be used in all network requests performed by the reconciler, as the integreatly-operator maintains this context
+ and may kill it if an uninstall is detected.
+ 
+##### installation 
+This is the CR we are basing the install from, it has values that are occasionally required by reconcilers, for example
+the namespace prefix. 
+
+##### product 
+This is a pointer to the this reconciler's product in the status block of the CR, it can be used to set values
+such as version, host and operator version.
+
+##### serverClient
+This is the client to the cluster, and is used for getting, creating, updating and deleting resources in the cluster.
+
+#### Return Values
+The return values from this method are `state` and `err`:
+##### State
+This is communicated back to the user via the status block of the RHMI CR, this is usually either in progress
```

Comment:
Could also mention here that the available options for this state are these 3 rhmi status phase types: https://github.com/integr8ly/integreatly-operator/blob/master/pkg/apis/integreatly/v1alpha1/rhmi_types.go#L24-L26

Reply:
done, thanks!